consensus/dbft/payload.go

```diff
@@ -215,25 +190,71 @@ func (t messageType) String() string {
 	}
 }
 
+// DecodeRLP decodes a message from RLP.
+func (m *message) DecodeRLP(s *rlp.Stream) error {
+	var em messageExt
+	if err := s.Decode(&em); err != nil {
+		return err
+	}
+	m.Type, m.BlockIndex, m.ValidatorIndex, m.ViewNumber = em.Type, em.BlockIndex, em.ValidatorIndex, em.ViewNumber
+	var obj interface{}
+	switch m.Type {
+	case changeViewType:
+		obj = &changeView{}
+	case prepareRequestType:
+		obj = &prepareRequest{}
+	case prepareResponseType:
+		obj = &prepareResponse{}
+	case commitType:
+		obj = &commit{}
+	case recoveryRequestType:
+		obj = &recoveryRequest{}
+	case recoveryMessageType:
+		obj = &recoveryMessage{}
+	default:
+		err := fmt.Errorf("invalid type: 0x%02x", byte(m.Type))
+		log.Warn(err.Error())
+		return err
+	}
+	if err := rlp.DecodeBytes(em.MsgPayload, obj); err != nil {
+		return err
+	}
+	if m.Type == changeViewType {
+		obj.(*changeView).newViewNumber = m.ViewNumber + 1
+	}
+	m.msgPayload = obj
+	return nil
```

Comment:
Will the following code work? Less change comparing to the previous non-rlp version.
```suggestion
	switch m.Type {
	case changeViewType:
		m.msgPayload = &changeView{
			// newViewNumber is not marshaled
			newViewNumber: m.ViewNumber + 1,
		}
	case prepareRequestType:
		m.msgPayload = new(prepareRequest)
	case prepareResponseType:
		m.msgPayload = new(prepareResponse)
	case commitType:
		m.msgPayload = &commit{}
	case recoveryRequestType:
		m.msgPayload = new(recoveryRequest)
	case recoveryMessageType:
		m.msgPayload = new(recoveryMessage)
	default:
		err := fmt.Errorf("invalid type: 0x%02x", byte(m.Type))
		return err
	}
	return rlp.DecodeBytes(em.MsgPayload, m.msgPayload)
```